client/lib/purchases/test/index.js

```diff
@@ -68,4 +72,40 @@ describe( 'index', () => {
 			expect( isCancelable( DOMAIN_PURCHASE_PENDING_TRANSFER ) ).to.be.false;
 		} );
 	} );
+	describe( '#isPaidWithCredits', () => {
+		test( 'should be true when paid with credits', () => {
+			expect( isPaidWithCredits( PLAN_PURCHASE_WITH_CREDITS ) ).to.be.true;
+		} );
+		test( 'should false when not paid with credits', () => {
+			expect( isPaidWithCredits( PLAN_PURCHASE_WITH_PAYPAL ) ).to.be.false;
+		} );
+		test( 'should be false when payment not set on purchase', () => {
+			const purchase = Object.assign( {}, DOMAIN_PURCHASE );
+			delete purchase.payment;
```

Comment:
If `payment` _is_ there, a nicer (non-mutating) way of doing this would be

```js
const { payment, ...purchase } = DOMAIN_PURCHASE;
```

Reply:
Thanks for the feedback. I rewrote the test so that it just passed `{}` and then tweaked the logic so that it returned `false` instead of `undefined`.